org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java

```diff
@@ -534,21 +548,21 @@ protected void _toJavaStatement(XTryCatchFinallyExpression expr, ITreeAppendable
 		b.decreaseIndentation().newLine().append("}");
 
 		// Catch and Finally
-		appendCatchAndFinally(expr, b, isReferenced);
+		appendCatchAndFinally(expr, b, isReferenced, throwablesStore);
 		b.closeScope();
 	}
 
-	protected void appendCatchAndFinally(XTryCatchFinallyExpression expr, ITreeAppendable b, boolean isReferenced) {
+	protected void appendCatchAndFinally(XTryCatchFinallyExpression expr, ITreeAppendable b, boolean isReferenced, String throwablesStore) {
```

Comment:
In general this PR looks good to me. I'm not entirely sure though if we should change the signature of this method or lookup the throwablesStore from the appendable again. @cdietrich what do you think?

Reply:
at least Jbase makes use of `appendCatchAndFinally` cc @LorenzoBettini

Reply:
this i think keeping the api might be better. but on the other hand i assume lorenzo has to update jbase for `?:` and try-with-resources anyway

Reply:
I do have a tendency towards API compatibility, too. So let's play this safe.

Reply:
I can also update Jbase according to the new API, though I'm also in favor of API compatibility ;)

what I probably missed is: conditional expression and try-with-resources are effectively part of Xbase? I seemed to understand that they were introduced in Xtend directly...

Reply:
yes its xtend. i missed that. sorry

Reply:
So I take the chance: why not in Xbase? ;)

Reply:
i assume this is https://github.com/eclipse/xtext-extras/issues/244 correct @szarnekow

Reply:
Indeed it is. Changing Xbase.ecore is possible in a backwards compatible way whereas changing the grammar unfortunately is not (as of now)

Reply:
but that means lorenzo could reuse ecore and co

Reply:
Exactly. Reusing the compiler, the ecore, the validations, the interpreter etc is possible. It's only the grammar that needs to be adjusted.